node/actors/bft/src/tests/unit_tests.rs

```diff
@@ -0,0 +1,599 @@
+    leader_prepare.sig = util.rng().gen();
+    let res = util.dispatch_leader_prepare(leader_prepare).await;
+
+    assert_eq!(
+        res,
+        Err(LeaderPrepareError::InvalidSignature(
+            SignatureVerificationFailure
+        ))
+    );
+}
+
+#[tokio::test]
+async fn leader_prepare_invalid_prepare_qc_different_views() {
+    let mut util = Util::make().await;
+
+    let replica_prepare = util.make_replica_prepare(None::<fn(&mut ReplicaPrepare)>);
+    let res = util.dispatch_replica_prepare(replica_prepare.clone());
+    assert_eq!(res, Ok(()));
+
+    let mut leader_prepare = util.recv_leader_prepare().await.unwrap();
+    leader_prepare.view = leader_prepare.view.next();
+    let leader_prepare = util
+        .own_key()
+        .sign_msg(ConsensusMsg::LeaderPrepare(leader_prepare));
+    let res = util.dispatch_leader_prepare(leader_prepare).await;
+
+    assert_eq!(
+        res,
+        Err(LeaderPrepareError::InvalidPrepareQC(anyhow!(
+            "PrepareQC contains messages for different views!"
+        )))
+    );
+}
+
+// #[tokio::test]
+// async fn replica_commit_sanity() {
+//     let mut util = Util::make().await;
+//
+//     let replica_prepare = util::make_replica_prepare(&consensus, None::<fn(&mut ReplicaPrepare)>);
+//     let res = util::dispatch_replica_prepare(&ctx, &mut consensus, replica_prepare);
+//     assert_eq!(res, Ok(()));
+//     let leader_prepare = util::make_leader_prepare_from_replica_prepare(&consensus, &mut rng, replica_prepare, Some(|msg: &mut LeaderPrepare| {
+//         msg.view = ViewNumber(2);
+//     }));
+//     let replica_commit = util::make_replica_commit(&consensus, leader_prepare.msg., None::<fn(&mut ReplicaCommit)>);
+//
+//     let res = scope::run!(&ctx, |ctx, s| {
+//         s.spawn_blocking(|| {
+//             let res = consensus
+//             .leader
+//             .process_replica_commit(ctx, &consensus.inner, leader_prepare.cast().unwrap());
+//             Ok(res)
+//         })
+//         .join(ctx)
+//     })
+//         .await
+//         .unwrap();
+//
+//     assert_eq!(
+//         res,
+//         Err(LeaderPrepareError::InvalidPrepareQC(anyhow!("PrepareQC contains messages for different views!")))
+//     );
+// }
+
+mod util {
+    use crate::{
+        io::{InputMessage, OutputMessage},
+        leader::ReplicaPrepareError,
+        replica::LeaderPrepareError,
+        Consensus,
+    };
+    use rand::{rngs::StdRng, Rng, SeedableRng};
+    use zksync_concurrency::{ctx, ctx::Ctx, scope};
+    use zksync_consensus_network::io::ConsensusInputMessage;
+    use zksync_consensus_roles::{
+        validator,
+        validator::{
+            BlockHeader, ConsensusMsg, LeaderPrepare, Payload, Phase, PrepareQC, ReplicaCommit,
+            ReplicaPrepare, SecretKey, Signed, ViewNumber,
+        },
+    };
+    use zksync_consensus_utils::pipe::DispatcherPipe;
+
+    pub(crate) struct Util {
```

Comment:
IMO, using the helper struct for tests is the right approach, and its interface makes sense 👍 (other than some minor details).